src/_pytest/python.py

```diff
@@ -1724,6 +1724,23 @@ def write_docstring(tw: TerminalWriter, doc: str, indent: str = "    ") -> None:
         tw.line(indent + line)
 
 
+class DeprecatingFuncArgs(Dict[str, object]):
+    def __init__(self, initialnames):
+        self.initialnames = initialnames
+        super().__init__()
+
+    def __getitem__(self, key: str) -> object:
+        if key not in self.initialnames:
+            warnings.warn(
+                "Access to names other than initialnames i.e., direct args,"
+                " the ones with `usefixture` or the ones with `autouse` through "
+                "`item.funcargs` is deprecated and will raise `KeyError` from "
+                "pytest 9. Please use `request.getfixturevalue` instead.",
+                DeprecationWarning,
```

Comment:
We don't use `DeprecationWarning` directly, instead see file  `src/_pytest/deprecated.py`, add `PytestRemovedIn9Warning`, and add a subclass of it.